src/e2e/sdk_remote_integration.spec.ts

```diff
@@ -0,0 +1,120 @@
+			RPCPort: opPort,
+			RPCHost: "0.0.0.0",
+			NodeReqTimeout: 10,
+			WaitMinedTimeout: 60
+		}), accept));
+
+		exec(`$ERDSTALL_EXE -config operator.cfg -epochs 10 -ledger-port ${nodePort} &`);
+		return new Promise(accept => setTimeout(accept, 10000));
+	});
+
+	function makeClient(index: number): Erdstall {
+		const provider = new ethers.providers.JsonRpcProvider(`http://localhost:${nodePort}`);
+		const derivationPath = `m/44'/60'/0'/0/${index+2}`;
+		const user = ethers.Wallet.fromMnemonic(mnemonic, derivationPath);
+		const userAddr = Address.fromString(user.address);
+		return sdk.NewClient(userAddr, user.connect(provider), opAddr);
+	}
+
+	let clients: Erdstall[] = [];
+
+	it("create clients", async() => {
+		for(let i = 0; i < 4; i++)
+			clients.push(makeClient(i));
+		
+		// establishes connection to erdstall-contract on ledger and connection
+		// to enclave with the given url. All event handlers were registered and
+		// forwarded were necessary.
+		return Promise.all(clients.map(x => x.initialize()));
+	});
+	it("subscribe clients", async() => Promise.all(
+		clients.map(c => c.subscribe())));
+
+	it("deposits", async() => {
+		const depositBal = new Assets();
+		depositBal.addAsset(assets.ETHZERO, new assets.Amount(100000000n));
+
+		const stages = await Promise.all(clients.map(c => c!.deposit(depositBal)));
+		return Promise.all(stages.map(stage => stage.wait()));
+	});
+
+	it("off-chain tx + await receipt", async () => {
+		const amount = new Assets();
+		amount.addAsset(assets.ETHZERO, new assets.Amount(100n));
+		await clients[0].transferTo(amount, clients[1].address);
+		return new Promise((resolve, reject) => {
+			let timeout = setTimeout(() => reject(new Error("Bob's receipt timed out")), 15000);
+			clients[1].once("receipt", (_: TxReceipt) => {
+				clearTimeout(timeout);
+				resolve();
+			});
+		});
+	});
+
+	it("off-chain transactions", async () => {
+		const txCount = 500;
+		const begin = Date.now();
+		let responses = [];
+		for(let i = 0; i < txCount; i++) {
+			const amount = new Assets();
+			amount.addAsset(assets.ETHZERO, new assets.Amount(BigInt(i)));
+			responses.push(clients[0].transferTo(amount, clients[1].address));
+		}
+		await Promise.all(responses);
+		const timeMs = (Date.now() - begin);
+		console.log(`Sent ${txCount} TXs in ${timeMs / 1000}s (${timeMs / txCount}ms/tx | ${1000 * txCount / timeMs}Hz)`);
+	});
+
+	it("leave", async() => {
+		const stages = await Promise.all(clients.slice(0,2).map(c => c.leave()));
+		return Promise.all(stages.map(stage => stage.wait()));
+	});
+
+	let nft: bigint;
+	it("lets users mint tokens", async() => {
+		nft = 89430923n;
+		return clients[2].mint(clients[2].address, nft);
+	});
+
+	it("lets users trade", async () => {
+		// Charlie trades his NFT for Dagobert's ETH.
+		const offer = new Assets();
+		offer.addAsset(clients[2].address, new assets.Tokens([nft]));
+		const expect = new Assets();
+		expect.addAsset(assets.ETHZERO, new assets.Amount(1000n));
```

Comment:
The `Assets` constructor now accepts a variadic list of `{address, Asset}`s that are added to it.